packages/snowflake/src/query-interface.ts

```diff
@@ -1,6 +1,63 @@
-import { AbstractQueryInterface } from '@sequelize/core';
+import { AbstractQueryInterface, QueryTypes } from '@sequelize/core';
 import type { SnowflakeDialect } from './dialect.js';
 
 export class SnowflakeQueryInterface<
   Dialect extends SnowflakeDialect = SnowflakeDialect,
-> extends AbstractQueryInterface<Dialect> {}
+> extends AbstractQueryInterface<Dialect> {
+  /**
+   * Ensure all required sequences are created for AUTOINCREMENT columns
+   *
+   * Snowflake doesn't support returning the last inserted ID for autoincrement columns.
+   * To overcome this, we create a sequence for each autoincrement column,
+   * and use it to get the next value.
+   *
+   * @param table  Table to create
+   * @param attributes Object representing a list of normalized table attributes
+   * @param [options]
+   *
+   * @protected
+   */
+  async ensureSequences(table: any, attributes: any, options: any) {
```

Comment:
Aren't there types for these that we can use?

Reply:
```suggestion
  async ensureSequences(table: TableOrModel, attributes: Record<string, string>, options: QueryRawOptions) {
```
Something like this should work I think. Haven't tested it out, especially the options type does need additional work later

Reply:
Thanks! Let me just change `table: TableOrModel` to `table: TableName`, as the caller seems to always pass just `TableName`. Otherwise, I'll need to add more weird logic to get the table from `Model`.